components/PreviewFrame/index.js

```diff
@@ -26,19 +26,28 @@ const screenSizes = {
   }
 }
 
-const PreviewFrame = ({ previewScreenSize, commitId, repoId, darkmode }) => {
+const PreviewFrame = ({
+  previewScreenSize,
+  commitId,
+  repoId,
+  darkmode,
+  sideBarWidth
+}) => {
   const [scaleFactor, setScaleFactor] = useState(1)
   const [leftSpace, setLeftSpace] = useState(0)
+  const [iframeLoading, setIframeLoading] = useState(true)
   const [colorScheme] = useColorContext()
   const iframeRef = useRef()
 
   const URL = `${PUBLIC_BASE_URL}/repo/${repoId}/preview?commitId=${commitId}&darkmode=${darkmode}`
```

Comment:
An absolute path (paths starting with `/` go from root) would also work here.

```suggestion
  const iframeSrc = `/repo/${repoId}/preview?commitId=${commitId}&darkmode=${darkmode}`
```
Btw. URL is a global class name—would avoid it. Also realised that `PUBLIC_BASE_URL` is missing in `.env.example`